### ISSUE TITLE
2.11b4: A menu item is not translated

### ISSUE THREAD

**Issue by mlep** [JabRef/jabref#159]
Just a glitch:
In JabRef 2.11b4 (but was present in beta3...), the menu item "File -> Save selected as plain BibTeX..." is not localized even after adding the key "Save_selected_as_plain_BibTeX...=[Translation here]" to the files Menu_XX.properties.

This menu item is defined in JabRefFrame.java (lines 249-252), but I could not figure out why it was not localized ; while the menu item located just above ("Save selected as ...") was localized...

**Comment by matthiasgeiger** [comment]
In c8af1565c6d7dde5ab2836d9282fbc00cea3fc0f I just added the key to the Menu_XX.properties files - works now for english and german in the master branch.
However, my language skills are not sufficient for the other languages :wink:

(... no rush to notify translators: there some more issues regarding untranslated menu items)

**Comment by simonharrer** [comment]
Maybe the python script that searches for Localization.lang does not work good enough to detect these keys? Can this be an issue?

**Comment by matthiasgeiger** [comment]
@mlep Just a short question to you as the "master of localization" :wink: What workflow do you prefer for missing keys? Should I directly add missing key to all Menu_XX.properties and JabRef_XX.properties files?
The python script seems to work one for JabRef_XX and all usages of `Localization.lang(...)` but not for `Localization.menuTitle(...)`

**Comment by koppor** [comment]
@matthiasgeiger Sure! Otherwise, the translators do not know which keys are required to be translated. I think, Menu entries have to be added by hand to `Menu_*.properties` as the script does not support updating these files.

**Comment by mlep** [comment]
@matthiasgeiger: Everybody is welcome to add missing keys to the *.properties files! On my side, I try to keep things tidy, and to send reminders to other translators. On this issue, I am considering myself more as a coordinator than a master ;-)

**Comment by koppor** [comment]
I'm not sure whether we should switch to a service like https://www.transifex.com/, but I don't know at all how it works.

**Comment by matthiasgeiger** [comment]
Good. I just added some few missing things... 

@koppor Rather expensive, isn't it?

**Comment by koppor** [comment]
@matthiasgeiger Quoting https://www.transifex.com/pricing/: "Haben Sie ein Open Source-Projekt? Crowdsourcen Sie die Lokalisierung Ihres Open-Source-Projektes kostenlos!"

**Comment by matthiasgeiger** [comment]
Okay... I missed this statement.

Back to topic: I will close this issue here, as the key has been already added to the property files.
However, I would create a new one for milestone v2.80 as a reminder to add all missing translations. 
Any objections?

**Comment by koppor** [comment]
@matthiasgeiger Could you do me the favor and copy the strings of `JabRef_*.properties` to `Menu_*.properties` in the `dev_2.11` branch?

The strings were indeed translated in 2.11b3. Did we change the resolution of these strings from b3 to b4? I don't know if we need these strings in `JabRef_*.properties` at all. I would vote for removing them completely there.

Excerpt of result of `grep -r "Save_selected_as" .` in `jabref-dev_2.11/src/main/resources/resource`
```
./JabRef_pt_BR.properties:Save_selected_as_...=Salvar_selecionados_como...
./JabRef_pt_BR.properties:Save_selected_as_plain_BibTeX_...=Salvar_selecionado_como_Bibtex_simples
./Menu_pt_BR.properties:Save_selected_as_...=Salvar_os_se&lecionados_como...
```

I can do the merging into the master branch and resolving the conflicts...

(Pro hint: [have two checkouts at the same time](https://git-scm.com/docs/git-worktree))

**Comment by matthiasgeiger** [comment]
The displayed menu text is not translated in 2.11b3 as there is no "Save_selected_as_plain_BibTeX_..." key in the `Menu_*.properties`. 
The key is present in the `JabRef_*.properties` and this key is used to translate the tooltip popping up when hovering over the menu title. 

(This is one weak point in the current localization: there are various strings which have to be maintained in two places:  `Menu_*.properties` and `JabRef_*.properties`)

But I have not seen, that the exactly same key is used for both translations so I will simply copy the ones from `JabRef_*.properties` to  `Menu_*.properties` (both in `master` and `dev_2.11`).

**Comment by simonharrer** [comment]
Would it make sense to have only a single properties file for all of the translations?

**Comment by matthiasgeiger** [comment]
Generally yes. 

Two drawbacks: 
1) For new translators a relatively small `Menu_XX.properties` files is a good starting point and not so daunting as the 2500+ lines `JabRef_xx.properties`.
2) Although the Strings for Menu and JabRef are almost equal, there is some special handling of Menu Strings as they might contain `&` for indicating Mnemonic keys. So there must be a solution to display the values correctly for menu and "normal" usage.

**Comment by mlep** [comment]
I agree with @matthiasgeiger and I can add more on this:
I do not think this is advisable for very practical reasons: when a new potential translator shows up, the menu_XX.properties is relatively short, you can easily see the results, and you get all the menus translated in the short term. So, it is an easy task, and you get rewarded soon. For motivation, this is a clear plus.
If we merge menu_XX.properties and JabRef_XX.properties, I am afraid it will discourage most candidates, and also it will be difficult to start by translating all the menu.
If there is a need to merge both files, I guess the option is to identify the strings related to the menus (for example, "Save_as...=Save_as..." could be changed into "MenuJabref_Save_as...=Save_as..."), so that we can tell translators "OK, the file is quite large, but do not be afraid: if you only do the strings starting with MenuJabref, it will be an easy and significant step forward.

**Comment by simonharrer** [comment]
Ok, then we should keep it as it is.

**Comment by matthiasgeiger** [comment]
@koppor This can be closed as you merged the translations back to master in 88a5cc3706e03f50977aae17515660cec247abb8?
